src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -294,10 +294,19 @@ function useChatUtilsHook(): UseChatUtilsHook {
     return pushMessage;
   };
 
+  const setSelectedChannel = (channel: ChannelList) => {
+    setChat((prevChat) => ({
+      ...prevChat,
+      selectedChannel: channel,
+      isLoadingFetchingChannelDetail: false
+    }));
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `setSelectedChannel` function is a good addition as it allows setting the selected channel in the chat state. However, it's important to ensure that the `channel` parameter passed to this function is always of type `ChannelList`. If there's a chance that this function could be called with an invalid or null value, you should add error handling or validation logic to prevent potential issues.



<!-- This is an auto-generated comment by OSS CodeRabbit -->